File: fundamentos/desafio-js/src/questions/higher-than-100-on-agency-33.ts
import { IAccount } from "../i-account";

/** O número de contas com mais de 100 reais de saldo na agência 33 é: */
export default function higherThan100OnAgency33(...accounts: Array<IAccount>) {
  const n = accounts
    .filter(acc => acc.balance > 100 && acc.agencia == 33)
    .length;
  console.info(`O número de contas com mais de 100 reais de saldo na agência 33 é: ${n}`);
}

File: fundamentos/desafio-dart/lib/questions/higher_than_100_on_agency_33.dart
import 'package:desafio_dart/domains/account.dart';

/// O número de contas com mais de 100 reais de saldo na agência 33 é:
void higherThan100OnAgency33(List<Account> accounts) {
  final n =
      accounts.where((acc) => acc.balance > 100 && acc.agencia == 33).length;

  print(
      'O número de contas com mais de 100 reais de saldo na agência 33 é: ${n}');
}
